gson/src/main/java/com/google/gson/TypeAdapter.java

```diff
@@ -191,13 +191,30 @@ public final void toJson(Writer out, T value) throws IOException {
           TypeAdapter.this.write(out, value);
         }
       }
+
+      @Override public JsonElement toJsonTree(T value) {
+        if (value == null) {
+          return JsonNull.INSTANCE;
+        } else {
+          return TypeAdapter.this.toJsonTree(value);
+        }
+      }
+
       @Override public T read(JsonReader reader) throws IOException {
         if (reader.peek() == JsonToken.NULL) {
           reader.nextNull();
           return null;
         }
         return TypeAdapter.this.read(reader);
       }
+
+      @Override public T fromJsonTree(JsonElement jsonTree) {
+        if (jsonTree.isJsonNull()) {
```

Comment:
@swankjesse The "if null" code was moved from the preceding read()/write() methods, not added.